coil-test/src/commonMain/kotlin/coil3/test/FakeImageLoaderEngine.kt

```diff
@@ -55,17 +55,17 @@ class FakeImageLoaderEngine private constructor(
 
     override suspend fun intercept(chain: Interceptor.Chain): ImageResult {
         val request = requestTransformer.transform(chain.request)
-        val requestValue = RequestValue(request, chain.size)
+        val newChain = chain.withRequest(request)
+        val requestValue = RequestValue(request, newChain.size)
         _requests.emit(requestValue)
 
         var result: ImageResult? = null
         for (interceptor in interceptors) {
-            val newChain = chain.withRequest(request)
```

Comment:
honestly I am not sure why a new chain was generated for each interceptor

Reply:
Agreed - this seems like a logic mistake.